flake.nix

```
@@ -29,9 +34,21 @@
         pkgs = pkgsFor.${system};
       in {
         default = pkgs.mkShell {
-          inputsFrom = [
-            packages.${system}.default
+          buildInputs = with pkgs; [
+            git
+            cmake
+            cargo-make
+            gnuplot
+            rustup
+            xz
+            wasm-pack
+            rust-bin.stable.latest.default
           ];
+          # Shared library liblzma.so.5 used by wasm-pack
+          shellHook = ''
+            xz_lib=$(nix-store -q --references $(which xz) | grep xz)
+            export LD_LIBRARY_PATH=$xz_lib/lib:$LD_LIBRARY_PATH
+          '';
```

I'm not sure this is necessary, `xz` should already include the `lib` in the `env`:
```
 > nix develop
warning: Git tree '/home/jakubgs/work/zerokit' is dirty

[jakubgs@caspair:~/work/zerokit]$ echo $NIX_LDFLAGS_FOR_TARGET
-L/nix/store/bmzls7bipx954xsnbgshh8kbi488zy5m-xz-5.6.3/lib -L/nix/store/3sal5rv8apvwmbki96b11fb2sl65d6hc-rust-default-1.86.0/lib

[jakubgs@caspair:~/work/zerokit]$ echo $CMAKE_LIBRARY_PATH
/nix/store/bmzls7bipx954xsnbgshh8kbi488zy5m-xz-5.6.3/lib:/nix/store/3sal5rv8apvwmbki96b11fb2sl65d6hc-rust-default-1.86.0/lib
```

---

Also, if that's not enough this can be simpler:
```suggestion
          shellHook = ''
            export LD_LIBRARY_PATH="${pkgs.xz.out}/lib:$LD_LIBRARY_PATH"
          '';
```